### ISSUE TITLE
Restrict per target?

### ISSUE THREAD

**Issue by ChojinDSL** [teryaev/drone-promote-auth#3]
Hi, the current implementation seems to limit the user to a specified branch, rather than promote target. Is this correct? Or am I doing something wrong?

Would it be possible to implement a feature to be able to limit users to a specific promotion target?

**Comment by teryaev** [comment]
Hi @ChojinDSL USER_PERMISSIONS param allows you to limit the access to specified targets per user. There's an example in readme:
```
--env=USER_PERMISSIONS=user4:uat;prod,user5:uat
```
in this case user4 is allowed to promote to uat and prod targets. user5 is only allowed to promote to uat.
I hope that helps.

**Comment by ChojinDSL** [comment]
Yes, I understand. But would it be possible to not restrict by branch, but rather by deployment target?

For example if in your .drone.yml you have a promote target:
event:
  - promote
 target: 
  - deploy-production
  
  so rather than restrict by branches, e.g. master, dev-test-1.0, etc...,
  the restriction applies to the promote target. So that only admins could promote a build to the "deploy-production" target.

**Comment by teryaev** [comment]
@ChojinDSL yes. This extension doesn't work with branches at all. By env I mean a promote target.

**Comment by ChojinDSL** [comment]
strange. In my tests it reacted only to branches. 
For example in my config for the plugin I set: 
USER_PERMISSIONS=ChojinDSL:docker

And the branch that I'm currently working in is called OPS-767.
If I try to promote a build to the target docker, I get the error that I'm not allowed to promote to OPS-767.

Then I changed the config to:
USER_PERMISSIONS=ChojinDSL:OPS-767
and I was able to promote to any promote target for builds in the branch OPS-767.

**Comment by ChojinDSL** [comment]
I'm using the latest drone docker image and my repo is bitbucket server. (In case that's relevant).
